indexing-hadoop/src/test/java/io/druid/indexer/IndexGeneratorJobTest.java

```diff
@@ -0,0 +1,344 @@
+                    "2014102200,i.example.com,963",
+                    "2014102200,j.example.com,333",
+                    "2014102300,a.example.com,100",
+                    "2014102300,b.exmaple.com,50",
+                    "2014102300,c.example.com,200",
+                    "2014102300,d.example.com,250",
+                    "2014102300,e.example.com,123",
+                    "2014102300,f.example.com,567",
+                    "2014102300,g.example.com,11",
+                    "2014102300,h.example.com,251",
+                    "2014102300,i.example.com,963",
+                    "2014102300,j.example.com,333"
+                )
+            },
+            {
+                "hashed",
+                "2014-10-22T00:00:00Z/P1D",
+                new Integer[][][]{
+                    {
+                        { 0, 4 },
+                        { 1, 4 },
+                        { 2, 4 },
+                        { 3, 4 }
+                    }
+                },
+                ImmutableList.of(
+                    "2014102200,a.example.com,100",
+                    "2014102201,b.exmaple.com,50",
+                    "2014102202,c.example.com,200",
+                    "2014102203,d.example.com,250",
+                    "2014102204,e.example.com,123",
+                    "2014102205,f.example.com,567",
+                    "2014102206,g.example.com,11",
+                    "2014102207,h.example.com,251",
+                    "2014102208,i.example.com,963",
+                    "2014102209,j.example.com,333",
+                    "2014102210,k.example.com,253",
+                    "2014102211,l.example.com,321",
+                    "2014102212,m.example.com,3125",
+                    "2014102213,n.example.com,234",
+                    "2014102214,o.example.com,325",
+                    "2014102215,p.example.com,3533",
+                    "2014102216,q.example.com,587"
+                )
+            }
+        }
+    );
+  }
+
+  private ObjectMapper mapper;
+  private HadoopDruidIndexerConfig config;
+  private File dataFile;
+  private File tmpDir;
+  private Interval interval;
+  private String partitionType;
+  private Object[][][] shardInfoForEachSegment;
+
+  public IndexGeneratorJobTest(
+      String partitionType,
+      String interval,
+      Object[][][] shardInfoForEachSegment,
+      List<String> data
+  ) throws IOException
+  {
+    this.partitionType = partitionType;
+    this.shardInfoForEachSegment = shardInfoForEachSegment;
+    this.interval = new Interval(interval);
+    mapper = new DefaultObjectMapper();
+    mapper.registerSubtypes(new NamedType(HashBasedNumberedShardSpec.class, "hashed"));
+    mapper.registerSubtypes(new NamedType(SingleDimensionShardSpec.class, "single"));
+    InjectableValues inject = new InjectableValues.Std().addValue(ObjectMapper.class, mapper);
+    mapper.setInjectableValues(inject);
+
+    dataFile = File.createTempFile("test_website_data", "tmp");
```

Comment:
Something like this:

`private final @Rule TemporaryFolder temporaryFolder = new TemporaryFolder();`

Then your temporary file is just

`File tmpFile = temporaryFolder.newFile();`

The rule properly handles all the cleanup